generators/server/templates/src/main/java/package/web/rest/_AccountResource.java

```diff
@@ -274,140 +282,149 @@ public User getAccount(Principal principal) {
-           throw new InvalidPasswordException();
-       }
-       userRepository.findOneByLogin(managedUserVM.getLogin().toLowerCase()).ifPresent(u -> {throw new LoginAlreadyUsedException();});
-       userRepository.findOneByEmailIgnoreCase(managedUserVM.getEmail()).ifPresent(u -> {throw new EmailAlreadyUsedException();});
-       User user = userService.registerUser(managedUserVM);
-       mailService.sendActivationEmail(user);
-   }
+    @PostMapping("/register")
+    @Timed
+    @ResponseStatus(HttpStatus.CREATED)
+    public void registerAccount(@Valid @RequestBody ManagedUserVM managedUserVM) {
+        if (!checkPasswordLength(managedUserVM.getPassword())) {
+            throw new InvalidPasswordException();
+        }
+        userRepository.findOneByLogin(managedUserVM.getLogin().toLowerCase()).ifPresent(u -> {throw new LoginAlreadyUsedException();});
+        userRepository.findOneByEmailIgnoreCase(managedUserVM.getEmail()).ifPresent(u -> {throw new EmailAlreadyUsedException();});
+        User user = userService.registerUser(managedUserVM);
+        mailService.sendActivationEmail(user);
+    }
 
-   /**
+    /**
     * GET  /activate : activate the registered user.
     *
     * @param key the activation key
     * @throws RuntimeException 500 (Internal Server Error) if the user couldn't be activated
     */
-   @GetMapping("/activate")
-   @Timed
-   public void activateAccount(@RequestParam(value = "key") String key) {
-       userService.activateRegistration(key).orElseThrow(RuntimeException::new);
-   }
+    @GetMapping("/activate")
+    @Timed
+    public void activateAccount(@RequestParam(value = "key") String key) {
+        Optional<User> user = userService.activateRegistration(key);
+        if (!user.isPresent()) {
+            throw new DefaultException("No user was found for this reset key.");
+        };
+    }
 
-   /**
+    /**
     * GET  /authenticate : check if the user is authenticated, and return its login.
     *
     * @param request the HTTP request
     * @return the login if the user is authenticated
     */
-   @GetMapping("/authenticate")
-   @Timed
-   public String isAuthenticated(HttpServletRequest request) {
-       log.debug("REST request to check if the current user is authenticated");
-       return request.getRemoteUser();
-   }
+    @GetMapping("/authenticate")
+    @Timed
+    public String isAuthenticated(HttpServletRequest request) {
+        log.debug("REST request to check if the current user is authenticated");
+        return request.getRemoteUser();
+    }
 
-   /**
+    /**
     * GET  /account : get the current user.
     *
     * @return the current user
     * @throws RuntimeException 500 (Internal Server Error) if the user couldn't be returned
     */
-   @GetMapping("/account")
-   @Timed
-   public UserDTO getAccount() {
-       return Optional.ofNullable(userService.getUserWithAuthorities())
-           .map(UserDTO::new)
-           .orElseThrow(RuntimeException::new);
```

Comment:
`.orElseThrow(() -> new DefaultException("The current logged in user could not be found"));` would be better

Reply:
yes but the main issue comes from the `.orElseThrow()` : as the return parameter is not used, Sonar is complaining. This is what I want to remove.

Reply:
No the return param is used here and Sonar doesn't complain

Reply:
yes I'm changing all those